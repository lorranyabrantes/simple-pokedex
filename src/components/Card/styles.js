import { StyleSheet } from "react-native";
import { SPACING, COLORS, BORDER_RADIUS } from "../../services/styles";

const styles = StyleSheet.create({
  container: {
    width: "33.3%",
    padding: SPACING.xsmall,
  },
  content: {
    borderWidth: 1,
    borderRadius: BORDER_RADIUS.medium,
    borderColor: COLORS.GRAY,
    padding: SPACING.small,
    justifyContent: "center",
    alignItems: "center",
  },
  name: {
    marginBottom: SPACING.xsmall,
  },
  image: {
    width: 35,
    height: 35,
  },
});

export default styles;
