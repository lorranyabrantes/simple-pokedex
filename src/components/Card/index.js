import React from "react";
import { View, Image } from "react-native";

import Button from "../Button";
import Text from "../Text";

import styles from "./styles";

const Card = ({ image, name, onPressButton }) => {
  return (
    <View testID="card" style={styles.container}>
      <View style={styles.content}>
        {image ? (
          <Image
            style={styles.image}
            source={{
              uri: image,
            }}
          />
        ) : null}

        <Text style={styles.name}>{name}</Text>
        <Button text={"detalhes"} onPress={onPressButton} />
      </View>
    </View>
  );
};

export default Card;
