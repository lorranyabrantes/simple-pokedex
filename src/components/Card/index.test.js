import React from "react";
import { render } from "@testing-library/react-native";

import Card from "./";

const props = {
  name: "Name",
  image: "",
  onPressButton: jest.fn(),
};

describe("Card", () => {
  test("should render correctly", async () => {
    const component = render(<Card {...props} />);

    expect(component.toJSON()).toMatchSnapshot();
  });
});
