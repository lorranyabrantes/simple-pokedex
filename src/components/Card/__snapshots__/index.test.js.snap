// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card should render correctly 1`] = `
<View
  style={
    Object {
      "padding": 4,
      "width": "33.3%",
    }
  }
  testID="card"
>
  <View
    style={
      Object {
        "alignItems": "center",
        "borderColor": "#CCCCCC",
        "borderRadius": 5,
        "borderWidth": 1,
        "justifyContent": "center",
        "padding": 8,
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#333333",
            "fontFamily": "Roboto-Regular",
            "fontSize": 14,
          },
          Object {
            "marginBottom": 4,
          },
        ]
      }
    >
      Name
    </Text>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#FECA1B",
          "borderRadius": 5,
          "height": 35,
          "justifyContent": "center",
          "opacity": 1,
          "paddingHorizontal": 16,
        }
      }
      testID="button"
    >
      <Text
        style={
          Array [
            Object {
              "color": "#333333",
              "fontFamily": "Roboto-Regular",
              "fontSize": 14,
            },
            undefined,
          ]
        }
      >
        detalhes
      </Text>
    </View>
  </View>
</View>
`;
